lib/Dialect/FIRRTL/Transforms/WireDFT.cpp

```diff
@@ -135,49 +139,101 @@ void WireDFTPass::runOnOperation() {
-      // Grab the enable value and remove the annotation.
-      enableSignal =
-          getValueByFieldID(ImplicitLocOpBuilder::atBlockBegin(
-                                module->getLoc(), module.getBodyBlock()),
-                            module.getArgument(i), anno.getFieldID());
-      enableModule = module;
-      return true;
-    });
-    if (error)
-      return signalPassFailure();
 
-    // Walk the module body looking for any operation marked as the DFT enable.
-    auto walkResult = module->walk([&](Operation *op) {
-      AnnotationSet::removeAnnotations(op, [&](Annotation anno) {
-        // If we have already encountered an error or this is not the right
-        // annotation kind, continue.
-        if (error || !anno.isClass(dftTestModeEnableAnnoClass))
-          return false;
+      // Check for enable signal annotation.
+      if (anno.isClass(dftTestModeEnableAnnoClass)) {
+        // If we have already found a DFT enable, emit an error.
         if (enableSignal) {
           auto diag =
-              op->emitError("more than one thing marked as a DFT enable");
+              module->emitError("more than one thing marked as a DFT enable");
           diag.attachNote(enableSignal.getLoc()) << "first thing defined here";
           error = true;
           return false;
         }
         // Grab the enable value and remove the annotation.
-        enableSignal = getValueByFieldID(
-            ImplicitLocOpBuilder::atBlockEnd(op->getLoc(), op->getBlock()),
-            op->getResult(0), anno.getFieldID());
+        enableSignal =
+            getValueByFieldID(ImplicitLocOpBuilder::atBlockBegin(
+                                  module->getLoc(), module.getBodyBlock()),
+                              module.getArgument(i), anno.getFieldID());
         enableModule = module;
         return true;
+      }
+
+      // Check for clock div bypass signal.
+      if (anno.isClass(dftClockDividerBypassAnnoClass)) {
+        // If we have already found a DFT clock div bypass signal, emit an
+        // error.
+        if (clockDivBypassSignal) {
+          module->emitError(
+                    "more than one thing marked as a DFT clock div bypass")
+                  .attachNote(clockDivBypassSignal.getLoc())
+              << "first thing defined here";
+          error = true;
+          return false;
+        }
+        clockDivBypassSignal =
+            getValueByFieldID(ImplicitLocOpBuilder::atBlockBegin(
+                                  module->getLoc(), module.getBodyBlock()),
+                              module.getArgument(i), anno.getFieldID());
+        clockDivBypassModule = module;
+        return true;
+      }
+
+      return false;
+    });
+    if (error)
+      return signalPassFailure();
+
+    // Walk the module body looking for any operation marked as the DFT enable.
+    auto walkResult = module->walk([&](Operation *op) {
+      AnnotationSet::removeAnnotations(op, [&](Annotation anno) {
```

Comment:
Is this lambda the exact same as the one above? Consider moving into a shared lambda.

Reply:
They're very similar, thanks for pushing.  Refactored!